pallets/manta-pay/src/lib.rs

```diff
@@ -0,0 +1,790 @@
+		SenderPostError, SenderPostingKey, SinkPostingKey, SourcePostingKey, TransferLedger,
+		TransferLedgerSuperPostingKey, TransferPostError,
+	},
+};
+use manta_crypto::{
+	constraint::ProofSystem,
+	merkle_tree::{self, forest::Configuration as _},
+};
+use manta_pay::config;
+use manta_primitives::{
+	assets::{FungibleLedger, FungibleLedgerConsequence},
+	types::{AssetId, Balance},
+};
+use manta_util::codec::Decode as _;
+use scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_std::{vec, vec::Vec};
+use types::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod test;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmark;
+
+pub mod weights;
+
+pub use pallet::*;
+
+/// Type Definitions for Protocol Structures
+pub mod types;
+
+/// MantaPay Pallet
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_runtime::traits::AccountIdConversion;
+
+	/// Extrinsic Weight Info
+	pub trait WeightInfo {
+		/// Returns the [`Weight`] of the [`Pallet::mint`] extrinsic.
+		fn mint() -> Weight;
+
+		/// Returns the [`Weight`] of the [`Pallet::private_transfer`] extrinsic.
+		fn private_transfer() -> Weight;
+
+		/// Returns the [`Weight`] of the [`Pallet::reclaim`] extrinsic.
+		fn reclaim() -> Weight;
+	}
+
+	/// Pallet
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// The module configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Fungible ledger
+		type FungibleLedger: FungibleLedger<Self>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// Pallet ID
+		type PalletId: Get<PalletId>;
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	/// Shards of the merkle tree
+	#[pallet::storage]
+	pub(super) type Shards<T: Config> =
+		StorageDoubleMap<_, Identity, u8, Identity, u64, (config::Utxo, EncryptedNote), ValueQuery>;
+
+	/// Shard trees
+	#[pallet::storage]
+	pub(super) type ShardTrees<T: Config> =
+		StorageMap<_, Identity, u8, UtxoMerkleTreePath, ValueQuery>;
+
+	/// Outputs of Utxo accumulator
+	#[pallet::storage]
+	pub(super) type UtxoAccumulatorOutputs<T: Config> =
+		StorageMap<_, Identity, config::UtxoAccumulatorOutput, (), ValueQuery>;
+
+	/// Utxo set
+	#[pallet::storage]
+	pub(super) type UtxoSet<T: Config> = StorageMap<_, Identity, config::Utxo, (), ValueQuery>;
+
+	/// Void number set
+	#[pallet::storage]
+	pub(super) type VoidNumberSet<T: Config> =
+		StorageMap<_, Identity, config::VoidNumber, (), ValueQuery>;
+
+	/// Void number set insertion order
+	#[pallet::storage]
+	pub(super) type VoidNumberSetInsertionOrder<T: Config> =
+		StorageMap<_, Identity, u64, config::VoidNumber, ValueQuery>;
+
+	/// Void number set size
```

Comment:
These comments can be improved.
Or just remove if they are the same as the code names.